bootloader/wscript

```diff
@@ -665,22 +665,48 @@ def configure(ctx):
 
     ## setup DEBUG environment
     ctx.setenv('debug', basic_env)  # Ensure env contains shared values.
+    set_win_link(ctx.env, False)
     debug_env = ctx.env
     # This defines enable verbose console output of the bootloader.
     ctx.env.append_value('DEFINES', ['LAUNCH_DEBUG'])
     ctx.env.append_value('DEFINES', 'NDEBUG')
 
     ## setup windowed DEBUG environment
     windowed('debugw', debug_env)
+    set_win_link(ctx.env, True)
 
     ## setup RELEASE environment
     ctx.setenv('release', basic_env)  # Ensure env contains shared values.
     release_env = ctx.env
+    set_win_link(ctx.env, False)
     ctx.env.append_value('DEFINES', 'NDEBUG')
 
     ## setup windowed RELEASE environment
     windowed('releasew', release_env)
-
+    set_win_link(ctx.env, True)
+
+def set_win_link(env, win):
+    if env.DEST_OS == 'win32' and env.CC_NAME == 'msvc':
+        b32 = env.PYI_ARCH=='32bit'
+        what = get_win_subsystem(win, b32)
+        env.append_value('LINKFLAGS', what)
+        _clean_win_subsystem(env)
+
+def get_win_subsystem(win, b32):
+    if win:
+        base = '/SUBSYSTEM:WINDOWS'
+    else:
+        base = '/SUBSYSTEM:CONSOLE'
+    if b32:
+       base += ',5.01'
+    else:
+       base += ',5.02'
+    return base
+    
+def _clean_win_subsystem(env):
+    for v in ['C','CXX']:
+        for plat in ['CONSOLE','NATIVE','POSIX','WINDOWS','WINDOWSCE']:
+            env[v+'FLAGS_'+plat] = []
```

Comment:
Please add a comment about what this does, why this is necessary and why this will not have any side-effects. 

Also the function name should be more spelling, pointing to win32 and msvc. Also for better understanding this function should be nested like `windowed()` is. Do we need a function at all?

Reply:
Use string-formating for better readability? Anyway, please follow PEP8: spaces.